### PR TITLE
java-microprofile: deprecate stack

### LINKED ISSUE #751: Mark java-microprofile stack as deprecated, replaced by java-openliberty
* Update stack.yaml
* description on `appsody list`
* message on run/debug/test

### PR DESCRIPTION
**Pull request by awisniew90**
### Checklist:

- [x ] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [ x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [ x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stacks-overview.md#stack-structure).

### Modifying an existing stack:

- [ x] Updated the stack version in `stack.yaml`

Deprecate the Java Microprofile Stack
- Update stack description to show deprecation
- Mark stack as deprecated in stack.yaml
- Indicate deprecation in Readme

### Related Issues:
Fixes #751

### CODE REVIEW

File: incubator/java-microprofile/stack.yaml [hunk 1]
```diff
@@ -18,3 +18,4 @@ requirements:
     appsody-version: ">= 0.5.0"
 templating-data:
     libertyversion: '19.0.0.12'
+deprecated: 04/14/2020
```
Comment (groeges) [inline incubator/java-microprofile/stack.yaml:21]: Would you be able to add some info text to the `deprecated` element stating why the stack was deprecated, as you have done in the description and the readme. The value of the `deprecated` element is displayed when an appsody init | run | test | debug | deploy | extract are executed, so having some additional info other than date might be useful.

Comment (groeges) [reply incubator/java-microprofile/stack.yaml:21]: We have requested the format to be used to be "<date> - <deprecation reason>"

Comment (awisniew90) [reply incubator/java-microprofile/stack.yaml]: Thanks @groeges - changes made.

**Comment by awisniew90** [comment]
@Kamran64  - This one is good to go when you get a change.  Thanks!

File: incubator/java-microprofile/stack.yaml [hunk 2]
```diff
@@ -1,20 +1,24 @@
 name: Eclipse MicroProfile®
-version: 0.2.25
-description: Eclipse MicroProfile on Open Liberty & OpenJ9 using Maven
+version: 0.2.26
+description: This stack has been deprecated. Please use the Open Liberty stack.
```
Comment (neeraj-laad) [inline incubator/java-microprofile/stack.yaml:3]: I think we should not be changing the stack description as that gets pulled in by `appsody list` and also on appsody website. We should keep the deprecation message only on the `deprecated` field.

Comment (awisniew90) [reply incubator/java-microprofile/stack.yaml]: Done

**Pull request merged**